### ISSUE TITLE
DomainError with `isapprox` for `ULogarithmic` arguments

### ISSUE THREAD

**Issue by stecrotti** [cjdoris/LogarithmicNumbers.jl#15]
Approximately comparing two `ULogarithmic` numbers `x,y` with `x<y` results in a `DomainError` because `norm(x-y)` is called by `isapprox` and the result of `x-y` is negative.
```julia
using LogarithmicNumbers
x = 0.8496911771040857
y = x + 1e-12
ex = exp(ULogarithmic, x)
ey = exp(ULogarithmic, y)
isapprox(ex, ey)
``` 
gives
```julia
ERROR: DomainError with (exp(0.8496911771040857), exp(0.8496911771050857)):
difference is negative
Stacktrace:
 [1] -(x::ULogFloat64, y::ULogFloat64)
   @ LogarithmicNumbers ~/.julia/packages/LogarithmicNumbers/wM2JA/src/LogarithmicNumbers.jl:493
 [2] isapprox(x::ULogFloat64, y::ULogFloat64; atol::Int64, rtol::Int64, nans::Bool, norm::typeof(abs))
   @ Base ./floatfuncs.jl:307
 [3] isapprox(x::ULogFloat64, y::ULogFloat64)
   @ Base ./floatfuncs.jl:304
 [4] top-level scope
   @ REPL[6]:1
```

**Comment by stecrotti** [comment]
A possible solution: swap the arguments if necessary
```julia
function Base.isapprox(x::T, y::T; kwargs...) where {T<:ULogarithmic}
    if x ≥ y
        return invoke(isapprox, Tuple{Real,Real}, x, y; kwargs...)
    else
        return invoke(isapprox, Tuple{Real,Real}, y, x; kwargs...)
    end
end
```

I tried it and it seems to work. It's the first time i use `invoke` though, not sure whether it's a safe thing to do

**Comment by cjdoris** [comment]
That solution looks fine to me - could you make a PR please?

**Comment by cjdoris** [comment]
Oh here's another way
```julia
function Base.isapprox(x::ULogarithmic, y::ULogarithmic; kwargs...)
    isapprox(Logarithmic(x), Logarithmic(y); kwargs...)
end
```
which avoids the need for `invoke`.

**Comment by cjdoris** [comment]
Fixed by #16.
